packages/devtools_app/test/visible_screens_test.dart

```diff
@@ -0,0 +1,183 @@
+import 'package:devtools_app/src/screen.dart';
+import 'package:devtools_app/src/service_manager.dart';
+import 'package:devtools_app/src/timeline/timeline_screen.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'support/mocks.dart';
+
+void main() {
+  group('visible_screens', () {
+    FakeServiceManager fakeServiceManager;
+
+    setUp(() async {
+      fakeServiceManager =
+          FakeServiceManager(useFakeService: true, availableLibraries: []);
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
+      setGlobal(FrameworkController, FrameworkController());
+
+      await serviceManager.isolateManager.selectedIsolateAvailable.future;
+    });
+
+    void setupMockValues({
+      bool web = false,
+      bool flutter = false,
+      bool profileMode = false,
+    }) {
+      mockIsDartVmApp(fakeServiceManager.connectedApp, !web);
+      if (web) {
+        fakeServiceManager.availableLibraries.add('dart:html');
+      }
+      mockIsFlutterApp(fakeServiceManager.connectedApp, flutter);
+      if (flutter) {
+        fakeServiceManager.availableLibraries
+            .add('package:flutter/src/widgets/binding.dart');
+      }
+      mockIsDebugFlutterApp(
+          fakeServiceManager.connectedApp, flutter && !profileMode);
+    }
+
+    testWidgets('are correct for Dart CLI app', (WidgetTester tester) async {
+      setupMockValues();
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            // InspectorScreen,
+            TimelineScreen,
+            MemoryScreen,
+            PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct for Dart Web app', (WidgetTester tester) async {
+      setupMockValues(web: true);
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            // InspectorScreen,
+            // TimelineScreen,
+            // MemoryScreen,
+            // PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct for Flutter (non-web) debug app',
+        (WidgetTester tester) async {
+      setupMockValues(flutter: true);
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            InspectorScreen,
+            TimelineScreen,
+            MemoryScreen,
+            PerformanceScreen,
+            DebuggerScreen,
+            NetworkScreen,
+            LoggingScreen,
+            if (codeSizeScreenEnabled) CodeSizeScreen,
+          ]));
+    });
+
+    testWidgets('are correct for Flutter (non-web) profile app',
+        (WidgetTester tester) async {
+      setupMockValues(flutter: true, profileMode: true);
+
+      expect(
+          visibleScreenTypes,
+          equals([
+            InspectorScreen,
```

Comment:
Inspector and debugger should be disabled for profile builds

Reply:
Done!